iis-workflows/iis-workflows-ingest-pmc/src/main/java/eu/dnetlib/iis/workflows/ingest/pmc/metadata/MetadataImporter.java

```diff
@@ -46,15 +85,29 @@ protected void map(AvroKey<DocumentText> key, NullWritable value, Context contex
 				saxParser.parse(new InputSource(
 						new StringReader(nlm.getText().toString())), 
 						pmcXmlHandler);
-                context.write(new AvroKey<ExtractedDocumentMetadata>(output.build()), 
-                		NullWritable.get());
+				mos.write(namedOutputMeta, 
+						new AvroKey<ExtractedDocumentMetadata>(output.build()));
         	 } catch (ParserConfigurationException e) {
-             	throw new IOException("Text extraction failed for id " + nlm.getId() + 
-             			" and text: " + nlm.getText(), e);
+        		 Map<CharSequence, CharSequence> auditSupplementaryData = new HashMap<CharSequence, CharSequence>();
+        			auditSupplementaryData.put(FAULT_TEXT, nlm.getText());
```

Comment:
I get your point. 

On the other hand, I don't like very flexible structures like this where you can throw almost everything into. I think that what you're doing here is trying to get a functionality of class inheritance from Object Oriented languages, i.e. make the structure open for extending. In OO in case of metadataextraction, we would have `MetadataExtractionFault` class that would inherit from `Fault` and add two additional fields: `processing_time` and `url` while in case of PMC ingester we would have `PMCIngestionFault` class that would inherit from `Fault` and add field `text`.

A solution that would provide (poor man's) inheritance might look like this:

    record BasicFaultInfo {
       string inputObjectId;
       long timestamp;
       string code;
       // (...)
    }

    record Fault {
       BasicFaultInfo core;
    }

    record MetadataExtractionFault {
       BasicFaultInfo core;
       int processing_time;
       string url;
    }

    record PMCIngestionFault {
       BasicFaultInfo core;       
       string text;
    }

`BasicFaultInfo` would be placed in a common `audit` namespace (just where `Fault` is now) while `MetadataExtractionFault` and `PMCIngestionFault` would be placed in namespaces corresponding to metadataextraction and pmcingestion, respectively.

A disadvantage of this approach is that it is more complex, however the advantages are:

- auxiliary fields are explicitly defined
- the auxiliary fields might be records
- auxiliary fields have proper types defined